web/experiments/feature_view.py

```diff
@@ -0,0 +1,283 @@
+    categ = pd.read_csv(norms_path, sep='\t')
+    categ = categ[['feature type', 'concept', 'feature']]
+
+    def get_feature_category(word_str):
+        """Get all feature category"""
+        word_str = word_str.replace('_', ' ')
+        return categ.loc[categ['feature'] == word_str]['feature type'].unique().astype(str)
+
+    def binsort_features(feature_f1):
+        """Partition by feature type eg. 'encycopedic'"""
+        cat_val = defaultdict(list)
+        for f in feature_f1.keys():
+            for c in get_feature_category(f):
+                cat_val[c].append(feature_f1[f])
+
+        return cat_val
+
+    bf = binsort_features(fs_id)
+
+    df = pd.DataFrame.from_dict(bf, orient='index')
+
+    logging.info('Current matplotlib backend is {}'.format(plt.get_backend()))
+
+    # workaround for running matplotlib image generation without Xserver
+    if not show_visual:
+        logging.info('Switching matplotlib backend to Agg')
+        plt.switch_backend('agg')
+
+    sns.set_style("whitegrid")
+
+    tdf = df.transpose()
+    tdf.sort_index(axis=1, inplace=True)
+
+    sns.boxplot(data=tdf, boxprops=dict(alpha=0.2))
+    ax = sns.swarmplot(data=tdf)
+    plt.title('CSLB {}'.format(fig_title))
+
+    ax.set(xlabel='Feature category', ylabel='Feature fit score')
+
+    if fig_path is not None:
+        ax.get_figure().savefig(fig_path)
+        logging.info('Figure saved: {}'.format(fig_path))
+
+    if show_visual:
+        plt.show()
+
+    plt.close()
+
+
+def _store_data(fs_id, fp_id, file_path):
+    score_df = pd.DataFrame(list(fs_id.items()), columns=['Feature', 'F1score'])
+
+    params_data = {'Feature': fp_id[:, 0], 'Alpha': fp_id[:, 1], 'EndIter': fp_id[:, 2], 'Intercept': fp_id[:, 3],
+                   'Coefs': list(fp_id[:, 4:])}
+
+    params_df = pd.DataFrame(params_data, columns=['Feature', 'Alpha', 'EndIter', 'Intercept', 'Coefs'])
+
+    df = score_df.merge(right=params_df, on='Feature')
+
+    df = df.set_index('Feature')
+    df.sort_index(inplace=True)
+    logging.info('Saving: {}'.format(file_path))
+    df.to_csv(file_path)
+
+
+def figure_from_csv(path, fig_title):
+    df = pd.read_csv(path, index_col=0)
+    d = dict(zip(df.index.tolist, df.F1score.values))
+
+    _generate_figure(fs_id=d, fig_title=fig_title)
+
+
+def cslb_experiment(embedding, cslb_path='./CSLB',
+                    save_path='./', figure_desc='', n_jobs=4, random_state=0, nb_hyper=20, max_iter=1800,
+                    show_dialog=False):
+    """
+    Evaluate how well embedding encode features perceptual features. This experiment use CSLB semantic norms.
+
+    Parameters
+    ----------
+    embedding: Embedding object
+        Loaded embedding
+
+    cslb_path: string
+        Path to folder with cslb files norms.dat and feature_matrix.dat
+
+    n_jobs: int
+        Numbers of threads used for learning
+
+    save_path: string
+        Path where figures and progress files will be saved
+
+    figure_desc: string
```

Comment:
Remove this param, it is very detailed. Less is more ;) When someone wants to adjust the figure, he will need to do much more

Reply:
I do not remove this. It is necessary if you want figures with names of embedding.

Reply:
Less is more :) I think it is way too detailed, so please remove. If someone wants pretty and customized plots, they will have to do it themselves.

Reply:
Let's split into two functions as we discussed.